x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/DateFieldScript.java

```diff
@@ -35,6 +35,27 @@
         DateFieldScript newInstance(LeafReaderContext ctx);
     }
 
+    public static final Factory PARSE_FROM_SOURCE = (field, params, lookup, formatter) -> (LeafFactory) ctx -> new DateFieldScript(
+        field,
+        params,
+        lookup,
+        formatter,
+        ctx
+    ) {
+        @Override
+        public void execute() {
+            for (Object v : extractFromSource(field)) {
+                if (v instanceof String) {
+                    try {
+                        emit(formatter.parseMillis((String) v));
```

Comment:
let's remove the parse function that's exposed for dates?

Reply:
I think it might still be required for Nik's tests?  I can try removing it and we'll see what CI says

Reply:
ok, I keep forgetting that the tests for runtime fields in the search request are still running. Let's address that next then, I mean updating the tests and removing the function.